synapse/lib/cell.py

```diff
@@ -1546,6 +1555,14 @@ async def shutdown(self, timeout=None):
         extra = await self.getLogExtra()
         logger.warning('Graceful shutdown initiated...', extra=extra)
 
+        # if we're the leader, lets see if we can handoff...
+        if self.isactive and self.ahaclient is not None:
+            peers = await self._getDemotePeers(timeout=timeout)
```

Comment:
Should we catch exceptions that could come from `reqAhaProxy` so that we continue shutting down?

Reply:
or maybe `_getDemotePeers` should just use `getAhaProxy` and return an empty peer list rather than raising?

Reply:
Actually... i think we may wanna be loud here since the issue is the AHA server needing an update. Lets discuss.